contributors/design-proposals/local-ephemeral-storage.md

```diff
@@ -0,0 +1,399 @@
+6. Primary partition health is monitored by an external entity like the “Node Problem Detector” which is expected to place appropriate taints.
+7. If a primary partition becomes unhealthy, the node is tainted and all pods running in it will be evicted by default, unless they tolerate that taint. Kubelet’s behavior on a node with unhealthy primary partition is undefined. Cluster administrators are expected to fix unhealthy primary partitions on nodes.
+
+### Bob runs batch workloads and is unsure of “storage” requirements
+
+1. Bob can create pods without any “storage” resource requirements.
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+     namespace: myns
+    spec:
+     containers:
+     - name: fooc
+       volumeMounts:
+       - name: myEmptyDir
+         mountPath: /mnt/data
+     volumes:
+     - name: myEmptyDir
+       emptyDir:
+    ```
+
+2. His cluster administrator being aware of the issues with disk reclamation latencies has intelligently decided not to allow overcommitting primary partitions. The cluster administrator has installed a [LimitRange](https://kubernetes.io/docs/user-guide/compute-resources/) to “myns” namespace that will set a default storage size. Note: A cluster administrator can also specify burst ranges and a host of other features supported by LimitRange for local storage.
+
+    ```yaml
+    apiVersion: v1
+    kind: LimitRange
+    metadata:
+      name: mylimits
+    spec:
+       - default:
+         ephemeral-storage: 2Gi
+         type: Container
+    ```
+
+3. The limit range will update the pod specification as follows:
+
+    ```yaml
+    apiVersion: v1
+    kind: pod
+    metadata:
+     name: foo
+    spec:
+     containers:
+     - name: fooc
+       resources:
+         limits:
+           ephemeral-storage: 2Gi
+    ```
+
+4. Bob’s “foo” pod can use upto “2Gi” for its containers. 
+5. If Bob’s pod “foo” exceeds the “default” storage limits and gets evicted, then Bob can set a minimum storage requirement for his containers and a higher `sizeLimit` for his EmptyDir volumes.
+
+  ```yaml
+  apiVersion: v1
+  kind: pod
+  metadata:
+   name: foo
+  spec:
+   containers:
+   - name: fooc
+     resources:
+       requests:
+         storage.kubernetes.io/logs: 500Mi
+         storage.kubernetes.io/overlay: 500Mi
+     volumeMounts:
+     - name: myEmptyDir
+       mountPath: /mnt/data
+   volumes:
+   - name: myEmptyDir
+     emptyDir:
+       sizeLimit: 2Gi
+  ```
+
+6. It is recommended to require `limits` to be specified for `storage` in all pods. `storage` will not affect the `QoS` Class of a pod since no SLA is intended to be provided for storage capacity isolation. It is recommended to use Persistent Volumes as much as possible and avoid primary partitions.
+
+### Alice manages a Database which needs access to “durable” and fast scratch space
```

Comment:
Should this be moved local persistent volume ?

Reply:
fixed